.gitlab/pipeline/check.yml

```diff
@@ -5,6 +5,7 @@ cargo-clippy:
     - .common-refs
     - .pipeline-stopper-artifacts
   script:
+    # The removal of `RUSTFLAGS` is related to the configuration of `cargo-clippy` in `.cargo/config.toml`.
```

Comment:
Can we be more specific here?

Reply:
also pinging @athei and @gilescope https://github.com/paritytech/substrate/pull/9694/files#diff-9a4f3e4537ebb7474452d131b0d969d89a51286f4269aac5ef268e712be17268R7

Reply:
If we've updated the compiler to 1.74 then there's a clippy PR that we can land that removes our usage of the hack in .cargo/config.toml as we can now store lints in the workspace toml file. I think in my PR there's a couple of new crates that need to have their Cargo.tomls tweaked and then it should run green. https://github.com/paritytech/polkadot-sdk/pull/2390

Reply:
@gilescope 
Giles, cool, thank you for the comment :)
Do you plan to finish #2390? If not, I could. Just, please, let me know.


CI has 1.74:
```
$ cargo --version
cargo 1.74.0 (ecb9851af 2023-10-18)
...
$ cargo +nightly --version
cargo 1.75.0-nightly (b4d18d4bd 2023-10-31)
```